Source/Fuse.Scripting/ScriptClass.uno

```diff
@@ -105,25 +112,107 @@ namespace Fuse.Scripting
 			_method = method;
 		}
 
+		[Obsolete("Use ScriptMethod<T>(string, Uno.Func<Fuse.Scripting.Context, T, object[], object)>) instead")]
+		public ScriptMethod(string name, Func<Context, T, object[], object> method, ExecutionThread thread): this(name, method, thread, true)
+		{
+		}
+
+		/** Create a ScriptMethod that will run on the script-thread
+
+			@param name Name of method
+			@param method The native implementation of the method
+		*/
+		public ScriptMethod(string name, Func<Context, T, object[], object> method): this(name, method, ExecutionThread.JavaScript, true)
+		{
+		}
+
+		[Obsolete("Use ScriptMethod<T>(string, Uno.Action<T)>), ScriptMethod<T>(string, Uno.Action<T, object[])>) or ScriptMethod<T>(string, Uno.Func<Fuse.Scripting.Context, T, object[], object)>) instead")]
 		public ScriptMethod(string name, Action<Context, T, object[]> method, ExecutionThread thread): base(name, thread)
 		{
 			if (method == null)
 				throw new ArgumentNullException(nameof(method));
 
+			_oldVoidMethod = method;
+		}
+
+		/** Create an argument-less ScriptMethod that will run on the UI-thread
+
+			@param name Name of method
+			@param method The native implementation of the method
+		*/
+		public ScriptMethod(string name, Action<T> method): base(name, ExecutionThread.MainThread)
+		{
+			if (method == null)
+				throw new ArgumentNullException(nameof(method));
+
 			_voidMethod = method;
 		}
 
+		/** Create a ScriptMethod that will run on the UI-thread
+
+			@param name Name of method
+			@param method The native implementation of the method
+		*/
+		public ScriptMethod(string name, Action<T, object[]> method): base(name, ExecutionThread.JavaScript)
+		{
+			if (method == null)
+				throw new ArgumentNullException(nameof(method));
+
+			_method = new ArgumentMirrorClosure<T>(method).Run;
+		}
+
+		class ArgumentMirrorClosure<T>
+		{
+			readonly Action<T, object[]> _action;
+			public ArgumentMirrorClosure(Action<T, object[]> action)
+			{
+				_action = action;
+			}
+
+			public object Run(Context c, T obj, object[] args)
+			{
+				var marahalledArguments = new List<object>();
```

Comment:
Is the marahall where nightmares (mara) live? ;P

Reply:
whups! Thanks for noticing ;)